packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -104,12 +189,118 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
+	 *
+	 * 3. If only `categories` or a `namespace` is provided, the event should match either one of them.
+	 *
+	 * 4. If a `namespace` pattern exception is specified in the {@link TelemetryFilter}, the event should not match the exception pattern.
+	 *
+	 * @param event - The telemetry event to check against the filters.
+	 *
+	 * @returns boolean `true` if the event matches any filter, otherwise `false`.
+	 */
 	private doesEventMatchFilter(event: ITelemetryBaseEvent): boolean {
 		for (const filter of this.config.filtering.filters ?? []) {
-			if (filter.category === event.category) {
+			if ("namespacePattern" in filter && filter.namespacePattern !== undefined) {
+				if (event.eventName.startsWith(filter.namespacePattern)) {
+					// Found matching namespace pattern, since filters are ordered in most specific first,
+					// this is the most specific, relevant matching filter for the event.
+
+					// By default, if no categories are defined then any category is a valid match.
+					let doesFilterCategoriesMatch = true;
+					if (
+						"categories" in filter &&
+						filter.categories !== undefined &&
+						filter.categories.length > 0
+					) {
+						doesFilterCategoriesMatch = false;
+						const matchingCategory = filter.categories.find(
+							(category) => category === event.category,
+						);
+						doesFilterCategoriesMatch = matchingCategory ? true : false;
+					}
+
+					if (doesFilterCategoriesMatch) {
+						// The most specific, relevant filter matches so no need to attempt to evaluate against other filters
+						// as long as the events namespace does not match any defined namespace exception.
+						if (filter.namespacePatternExceptions !== undefined) {
+							for (const patternException of filter.namespacePatternExceptions) {
+								if (event.eventName.startsWith(patternException)) {
+									return false;
+								}
+							}
+						}
+						return true;
+					} else {
+						return false;
+					}
+				}
+			}
+			// Filter only has categories defined
+			else if (
+				"categories" in filter &&
+				filter.categories !== undefined &&
+				filter.categories.length > 0
+			) {
+				const doesFilterCategoriesMatch = filter.categories.find(
+					(category) => category === event.category,
+				);
+
+				// This filter specified no namespaces but it has a category match.
+				// Since filters are ordered by most specific first, we know that no previous
+				// filters with namespaces matched so we can return true.
+				if (doesFilterCategoriesMatch) {
+					return true;
+				} else {
+					continue;
+				}
+			} else {
 				return true;
 			}
 		}
+
 		return false;
 	}
+
+	/**
+	 * Checks an array of telemetry filters for any issues, merges redundant filters, and returns a fully validated array.
+	 *
+	 * @throws - an Error if there are two filters with duplicate namespace patterns or a filter with a pattern exception that is not a child of the parent pattern.
```

Comment:
```suggestion
	 * @throws An error if there are two filters with duplicate namespace patterns or a filter with a pattern exception that is not a child of the parent pattern.
```

Reply:
Generally, only the `@param` comments want a `-` in their syntax. Including a hyphen here will result in it appearing as a string literal in the docs.

Reply:
Updated.